app/src/main/java/io/github/polysmee/profile/fragments/ProfileActivityInfosFragment.java

```diff
@@ -12,104 +12,159 @@
 import com.google.firebase.auth.FirebaseUser;
 
 import io.github.polysmee.R;
+import io.github.polysmee.database.DatabaseUser;
+import io.github.polysmee.database.User;
 import io.github.polysmee.database.databaselisteners.StringValueListener;
 import io.github.polysmee.login.AuthenticationFactory;
 import io.github.polysmee.login.MainUser;
 import io.github.polysmee.profile.FriendsActivity;
 import io.github.polysmee.profile.ProfileActivity;
-import io.github.polysmee.settings.UserInfoDataStore;
+import io.github.polysmee.settings.MainUserInfoDataStore;
 
-public class ProfileActivityInfosFragment extends PreferenceFragmentCompat {
+public final class ProfileActivityInfosFragment extends PreferenceFragmentCompat {
 
     private StringValueListener nameListener;
-    private UserInfoDataStore userInfoDataStore;
+    private StringValueListener descriptionListener;
     private String visitingMode;
 
     @Override
+    // There is 1 argument to pass by the Bundle, and 1 optional argument if the value of PROFILE_VISIT_CODE needed argument is PROFILE_VISITING_MODE
     public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
         Context context = getPreferenceManager().getContext();
         PreferenceScreen screen = getPreferenceManager().createPreferenceScreen(context);
         visitingMode = this.getArguments().getString(ProfileActivity.PROFILE_VISIT_CODE);
-        FirebaseUser user = AuthenticationFactory.getAdaptedInstance().getCurrentUser();
-        if (visitingMode.equals(ProfileActivity.PROFILE_VISITING_MODE))
-            userInfoDataStore = new UserInfoDataStore(this.getArguments().getString(ProfileActivity.PROFILE_ID_USER));
-        else
-            userInfoDataStore = new UserInfoDataStore();
+        FirebaseUser mainUser = AuthenticationFactory.getAdaptedInstance().getCurrentUser();
```

Comment:
Here I don't know if it's possible or makes sense but if possible it would be better to use MainUser.getMainUser() instead to avoid exposing the specific database we use outside of the Database package. I noticed the User interface doesn't have a getEmail() method so maybe it isn't possible but it's something to keep in mind.

Reply:
Actually, there is a MainUser.getCurrentUserEmail() method, so this is totally possible.

Reply:
Indeed I didn't notice it, I'm going to fix it.